src/snowflake/snowpark/mock/_stored_procedure.py

```diff
@@ -54,20 +67,51 @@ def __call__(
         for arg, expected_type in zip(args, self._input_types):
             if isinstance(arg, Column):
                 expr = arg._expression
-                if not sproc_types_are_compatible(expr.datatype, expected_type):
+
+                # If expression does not define its datatype we cannot verify it's compatibale.
+                # This is potentially unsafe.
+                if expr.datatype and not sproc_types_are_compatible(
```

Comment:
Question: in what case is expr.datatype empty here?

Reply:
The datatype of the sqrt function in this example is None:
https://github.com/snowflakedb/snowpark-python/pull/1338/files#diff-4963488bfdaad538e6af7222d4db977758e66d05b4c11b600583449152416f50L221